src/app/services/reference-data/reference-data.service.spec.ts

```diff
@@ -22,66 +36,95 @@ describe('TechnicalRecordService', () => {
 
   describe('API', () => {
     describe('resourceTypes', () => {
-      it.each(testCases)('should return all data for a given resourceType', (value, done: any) => {
-        service.fetchReferenceData(value.resourceType).subscribe((response) => {
-          expect(response).toEqual(value.payload);
-          done();
+      it.each(testCases)('should return all data for a given resourceType', value => {
+        const apiResponse: ReferenceDataApiResponse = { data: value.payload };
+        service.fetchReferenceData(value.resourceType).subscribe(response => {
+          expect(response).toEqual(apiResponse);
         });
+
+        const req = controller.expectOne('https://url/api/v1/reference/COUNTRY_OF_REGISTRATION');
+        req.flush(apiResponse);
       });
     });
 
-    it('should thrown an error if resource type is not given', (done) => {
+    it('should thrown an error if resource type is not given', done => {
```

Comment:
If we're adding negative testing then theres the resource type not valid

Reply:
The API returns a 200 with an empty "data" array. So the test is not representative of what would actually happen if you called the API with a key that does not exist. I believe that was the reason behind it.